src/Centreon/Application/Controller/Monitoring/MetricController.php

```diff
@@ -0,0 +1,140 @@
+use FOS\RestBundle\View\View;
+use Centreon\Domain\Exception\EntityNotFoundException;
+
+/**
+ * This class is design to manage all API REST about metric requests
+ *
+ * @package Centreon\Application\Controller\Metric\Controller
+ */
+class MetricController extends AbstractController
+{
+    /**
+     * @var MetricServiceInterface
+     */
+    private $metricService;
+
+    /**
+     * @var MonitoringServiceInterface
+     */
+    private $monitoringService;
+
+    /**
+     * MetricController constructor.
+     *
+     * @param MetricServiceInterface $metricService
+     */
+    public function __construct(
+        MetricServiceInterface $metricService,
+        MonitoringServiceInterface $monitoringService
+    ) {
+        $this->metricService = $metricService;
+        $this->monitoringService = $monitoringService;
+    }
+
+    /**
+     * Entry point to get service metrics
+     *
+     * @param int $hostId
+     * @param int $serviceId
+     * @return View
+     * @throws \Exception
+     */
+    public function getServiceMetrics(
+        int $hostId,
+        int $serviceId,
+        \DateTime $start,
+        \DateTime $end
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+
+        /**
+         * @var $contact Contact
+         */
+        $contact = $this->getUser();
+        $this->monitoringService->filterByContact($contact);
+
+        $host = $this->monitoringService->findOneHost($hostId);
+        if (is_null($host)) {
+            throw new EntityNotFoundException('Host not found');
+        }
+
+        $service = $this->monitoringService->findOneService($hostId, $serviceId);
+        if (is_null($service)) {
+            throw new EntityNotFoundException('Service not found');
+        }
+        $service->setHost($host);
+
+        $metrics = $this->metricService
+            ->filterByContact($contact)
+            ->findMetricsByService($service, $start, $end);
+
+        return $this->view($metrics);
+    }
+
+    /**
+     * Entry point to get service status
+     *
+     * @param int $hostId
+     * @param int $serviceId
+     * @return View
+     * @throws \Exception
+     */
+    public function getServiceStatus(
+        int $hostId,
+        int $serviceId,
+        \DateTime $start,
+        \DateTime $end
+    ): View {
+        $this->denyAccessUnlessGrantedForApiRealtime();
+
+        /**
+         * @var $contact Contact
+         */
+        $contact = $this->getUser();
+        $this->monitoringService->filterByContact($contact);
+
+        $host = $this->monitoringService->findOneHost($hostId);
+        if (is_null($host)) {
+            throw new EntityNotFoundException('Host not found');
+        }
+
+        $service = $this->monitoringService->findOneService($hostId, $serviceId);
+        if (is_null($service)) {
+            throw new EntityNotFoundException('Service not found');
+        }
+        $service->setHost($host);
+
```

Comment:
as at the end, you want to find the right service that matches the given criteria, the same block code for the different methods. it could be "recfactored" in a single utility method ?